simple_https_client/src/lib.rs

```diff
@@ -197,13 +197,7 @@ where
     // Initialize TLS 1.3 client.
     let (client_hello, client_state) = {
         let sni = Bytes::from(host.as_bytes());
-        let ent = {
-            let mut entropy = [0u8; 64];
-            thread_rng().fill(&mut entropy);
-            Entropy::from(&entropy)
-        };
-
-        Client::connect(algorithms, &sni, None, None, ent)?
+        Client::connect(algorithms, &sni, None, None, &mut thread_rng())?
```

Comment:
I would prefer to keep the old style of providing a fixed amount of randomness to the client and server code so that we do not make the tls13 code observably stateful and harder to verify. I also think the TLS API is a bit more flexible for use in (say) a WASM environment if we do not hardcode the need for an external RNG. If the design decision is enforced by the underlying libcrux bindings, we can see how to make it work.

Reply:
We need randomness for libcrux. And I don't think we want to add an API to libcrux for this. Any real code would need this and I think this is a good place to figure out how to properly do this.
We ignored the entropy so far and created a new rng in the crypto module for libcrux. One option would be to isolate the crypto module and handle randomness properly in there. But I think we'll need to learn how to handle code like this anyway.

Reply:
Resolving for now. Let's rework this later if needed.